arshal_default.go

```diff
@@ -1015,7 +1050,13 @@ func makeStructArshaler(t reflect.Type) *arshaler {
 			// Write the object member value.
 			flagsOriginal := mo.Flags
 			if f.string {
-				mo.Flags.Set(jsonflags.StringifyNumbers | 1)
+				if mo.Flags.Get(jsonflags.StringifyWithLegacySemantics) {
+					if canLegacyStringify(f.typ) {
+						mo.Flags.Set(jsonflags.StringifyNumbers | jsonflags.StringifyBoolsAndStrings | 1)
+					}
+				} else {
+					mo.Flags.Set(jsonflags.StringifyNumbers | 1)
+				}
 			}
```

Comment:
Anyway we can reduce the indentation here?

Reply:
we have more indentation elsewhere in the same func so I think this is somewhat consistent. these days I'm OK with reasonably large or indented funcs unless there are pieces which can logically be split up well, I wouldn't split just for the sake of avoiding indentation.

Reply:
Semantically this is:
```go
if CondA1 {
    if CondA2 {
        ExprA()
    }
} else {
    ExprB()
}
```
which I guess can be rewritten as:
```go
if !CondA1 {
    ExprB()
} else if CondA2 {
    ExprA()
}
```